extensions/amp-story/1.0/amp-story-reaction-quiz.js

```diff
@@ -97,32 +96,22 @@ export class AmpStoryReactionQuiz extends AmpStoryReaction {
    * @param {Element} root
    */
   attachContent_(root) {
-    // TODO(jackbsteinberg): Optional prompt behavior must be implemented here
-    const promptInput = this.element.children[0];
     const promptContainer = root.querySelector(
       '.i-amphtml-story-reaction-quiz-prompt-container'
     );
 
-    // First child must be heading h1-h3
-    if (!['h1', 'h2', 'h3'].includes(promptInput.tagName.toLowerCase())) {
-      devAssert(
-        false,
-        'The first child must be a heading element <h1>, <h2>, or <h3>'
-      );
+    if (!this.element.hasAttribute('prompt-text')) {
       this.rootEl_.removeChild(promptContainer);
     } else {
-      const prompt = document.createElement(promptInput.tagName);
+      const prompt = document.createElement('h1');
 
-      prompt.textContent = promptInput.textContent;
+      prompt.textContent = this.element.getAttribute('prompt-text');
       prompt.classList.add('i-amphtml-story-reaction-quiz-prompt');
-
-      this.element.removeChild(promptInput);
       promptContainer.appendChild(prompt);
     }
 
-    const options = toArray(this.element.querySelectorAll('option'));
     devAssert(
-      options.length >= 2 && options.length <= 4,
+      this.options_ && this.options_.length >= 2 && this.options_.length <= 4,
```

Comment:
Please improve the error message on the next line to say that it expected 2 to 4 options but got X. If a publisher writes `option-9-text="..."` but only has 3 options, it'll give them a hint at what they got wrong and how to fix it.

Also: we want each component to have this assertion, and we want the code to fail even in production when this happens. Wdyt of having each concrete class provide their own constant for how many options they allow, and have this assertion in the abstract class?

Reply:
Added this.optionBounds_ as a protected variable that can be overriden from the concrete classes. On parseOptions I make sure the bounds are met, otherwise it will dev.error. Also improved the error messages.